src/Dockerfile

```diff
@@ -38,19 +38,26 @@ RUN <<EOT
 set -ex
 if ! yum install -y epel-release; then
   if . /etc/os-release 2>/dev/null; then
-    if [ "$ID" != "fedora" ]; then
-      yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-${VERSION:0:1}.noarch.rpm
-    fi
+    case "$ID" in
+      fedora) ;;
+      rocky)
+        dnf install -y epel-release
+        ;;
+      *)
+        yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-${VERSION:0:1}.noarch.rpm
+        yum update -y
+        ;;
+    esac
   fi
 fi
+if command -v dnf >/dev/null 2>/dev/null; then
+  dnf install -y bats vim
+else
+  yum install -y bats vim
```

Comment:
follow-up nit: would be good to make it so that the package list is not duplicated so multiple lines don't need to be changed for modifications and they can't accidentally diverge (at least until both use same package names). Eg. can put the arg0 to a variable.